spyder/api/plugins.py

```diff
@@ -533,3 +557,874 @@ def set_ancestor(self, ancestor):
+    # ]
+    CONF_DEFAULTS = None
+
+    # Define configuration version if using a separate file
+    #
+    # IMPORTANT NOTES:
+    # 1. If you want to *change* the default value of a current option, you
+    #    need to do a MINOR update in config version, e.g. from 3.0.0 to 3.1.0
+    # 2. If you want to *remove* options that are no longer needed or if you
+    #    want to *rename* options, then you need to do a MAJOR update in
+    #    version, e.g. from 3.0.0 to 4.0.0
+    # 3. You don't need to touch this value if you're just adding a new option
+    CONF_VERSION = None
+
+    # Widget to be used as entry in Spyder Preferences dialog
+    CONF_WIDGET_CLASS = None
+
+    # Path for images relative to the plugin path
+    IMG_PATH = 'images'
+
+    # Control the font size relative to the global fonts defined in Spyder
+    FONT_SIZE_DELTA = 0
+    RICH_FONT_SIZE_DELTA = 0
+
+    # --- API: Signals -------------------------------------------------------
+    # ------------------------------------------------------------------------
+
+    # Use this signal to inform another plugin that configuration changed.
+    sig_free_memory_requested = Signal()
+    sig_option_changed = Signal(str, object)
+    sig_quit_requested = Signal()
+    sig_redirect_stdio_requested = Signal(bool)
+    sig_restart_requested = Signal()
+    sig_shortcut_registration_requested = Signal(QObject, str, str, bool)
+    sig_status_message_requested = Signal(str, int)
+
+    # --- Private attributes -------------------------------------------------
+    # ------------------------------------------------------------------------
+    # Define configuration name map for plugin to split configuration
+    # among several files. See spyder/config/main.py
+    _CONF_NAME_MAP = None
+
+    # Store reference to registered plugins in a dictionary
+    # {plugin.NAME: plugin_instance, ...}
+    _PLUGINS = None
+
+    def __init__(self, parent, configuration=None):
+        super().__init__(parent)
+
+        self._main = parent
+        self.main = parent
+        self._conf = configuration
+        self._widget = None
+        self._plugin_path = os.path.dirname(inspect.getfile(self.__class__))
+        self._is_registered = None
+        self._is_compatible = None
+
+        self._actions = {}
+
+    # --- Private methods ---------------------------------------------
+    # ------------------------------------------------------------------------
+    def _register(self):
+        """
+        Setup and register plugin in Spyder's main window and connect it to
+        other plugins.
+        """
+        self.WIDGET_CLASS if self.WIDGET_CLASS else None
+
+        # Checks
+        # --------------------------------------------------------------------
+        if self._PLUGINS is None:
+            self._PLUGINS = {}
+
+        if self.NAME is None:
+            raise Exception('A Spyder Plugin must define a `NAME`!')
```

Comment:
Why are you using such a broad exception here? Perhaps `RuntimeError` would be better?

Reply:
Agreed, all the exceptions are currently broad and need to be updated.